src/components/TrackList.test.js

```diff
@@ -0,0 +1,168 @@
+
+        expect(queryByTestId("delete-button-component1")).toBeTruthy();
+        
+    });
+
+    it('should call onChange when a track is changed', async () => {
+        const fakeOnChange = jest.fn();
+        const fakeOnDelete = jest.fn();
+        const { queryByTestId, getByText, rerender } = render(
+            <TrackList
+                tracks={tracks}
+                availableTracks={availableTracks}
+                availableColors={availableColors}
+                onChange={fakeOnChange}
+                onDelete={fakeOnDelete}
+            />
+        );
+
+        // select file type
+        const fileTypeSelectComponent = queryByTestId("file-type-select-component1");
+        fireEvent.keyDown(fileTypeSelectComponent.firstChild, {key: "ArrowDown"});
+        await waitFor(() => getByText("haplotype"));
+        fireEvent.click(getByText("haplotype"));
+
+        // should wait for file to be selected
+        expect(fakeOnChange).toHaveBeenCalledTimes(0);
+
+        // simulate dropdown effect
+        let newTracks = {...tracks};
+        newTracks[1].trackType = "haplotype";
+
+        rerenderTrackList(rerender, newTracks, fakeOnChange, fakeOnDelete);
+
+        // change file name
+        const fileSelectComponent = queryByTestId('file-select-component1');
+        fireEvent.keyDown(fileSelectComponent.firstChild, {key: "ArrowDown"});
+        await waitFor(() => getByText("fileB1.gbwt"));
+        fireEvent.click(getByText("fileB1.gbwt"));
+
+        expect(fakeOnChange).toHaveBeenCalledTimes(1);
+        newTracks[1].trackFile = {"name": "fileB1.gbwt", "type": "haplotype"};
+        expect(fakeOnChange).toHaveBeenCalledWith(newTracks);
+
+        // rerender file select effect
+        rerenderTrackList(rerender, newTracks, fakeOnChange, fakeOnDelete);
+
+        // change color
+        const settingsButtonComponent = queryByTestId('settings-button-component1');
+        fireEvent.click(settingsButtonComponent);
+        await waitFor(() => getByText("reds"));
+        fireEvent.click(getByText("reds"));
+        fireEvent.click(document);
+
+        expect(fakeOnChange).toHaveBeenCalledTimes(2); 
+
+        newTracks[1].trackColorSettings.mainPallete = "reds";
+        expect(fakeOnChange).toHaveBeenCalledWith(newTracks);
+
+    });
+
+    it('should use a new event handler when passed', async () => {
+        const fakeOnChange1 = jest.fn();
+        const fakeOnChange2 = jest.fn();
+        const fakeOnDelete = jest.fn();
+        const { queryByTestId, getByText, rerender } = render(
+            <TrackList
+                tracks={tracks}
+                availableTracks={availableTracks}
+                availableColors={availableColors}
+                onChange={fakeOnChange1}
+                onDelete={fakeOnDelete}
+            />
+        );
+
+        expect(fakeOnChange1).toHaveBeenCalledTimes(0); 
+
+        let newTracks = {...tracks};
+        newTracks[1].trackType = "graph"
```

Comment:
This doesn't do anything, right? The track type for that entry is already `"graph"`.